src/imap_stream.rs

```diff
@@ -0,0 +1,195 @@
+use std::pin::Pin;
```

Comment:
do you mind adding a module doc string describing what imap_stream is for, and its properties?